FlowCrypt/src/main/java/com/flowcrypt/email/api/email/gmail/GmailHistoryHandler.kt

```diff
@@ -388,21 +394,34 @@ object GmailHistoryHandler {
             historyMessageLabelIds.joinToString(LABEL_IDS_SEPARATOR)
           )
 
-          if (localFolder.fullName in (historyLabelRemoved.labelIds ?: emptyList())) {
-            newCandidatesMap.remove(historyLabelRemoved.message.uid)
-            updateCandidates.remove(historyLabelRemoved.message.uid)
-            deleteCandidates[historyLabelRemoved.message.uid] = historyLabelRemoved.message.threadId
-            continue
-          }
-
-          if (LABEL_TRASH in (historyLabelRemoved.labelIds ?: emptyList())) {
-            val message = historyLabelRemoved.message
-            if (localFolder.fullName in historyMessageLabelIds) {
-              deleteCandidates.remove(message.uid)
-              updateCandidates.remove(message.uid)
-              newCandidatesMap[message.uid] = message
+          if (localFolder.isAll) {
+            if (
+              historyLabelRemoved.labelIds?.contains(LABEL_TRASH) == true ||
+              historyLabelRemoved.labelIds?.contains(LABEL_SPAM) == true
+            ) {
+              deleteCandidates.remove(historyLabelRemoved.message.uid)
+              updateCandidates.remove(historyLabelRemoved.message.uid)
+              newCandidatesMap[historyLabelRemoved.message.uid] = historyLabelRemoved.message
               continue
             }
+          } else {
+            if (localFolder.fullName in (historyLabelRemoved.labelIds ?: emptyList())) {
```

Comment:
Small comment. Let's add `lalelIds` variable in upcoming PR as `historyLabelRemoved.labelIds ?: emptyList<String>()` one is used several times 
```kt
val labelIds = historyLabelRemoved.labelIds ?: emptyList<String>()
```